sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -956,9 +957,12 @@ class CmisManagerTask:
     CMIS_STATE_REMOVED   = 'REMOVED'
     CMIS_STATE_FAILED    = 'FAILED'
 
-    def __init__(self, namespaces, port_mapping, skip_cmis_mgr=False):
-        self.task_stopping_event = multiprocessing.Event()
-        self.task_process = None
+    def __init__(self, namespaces, port_mapping, main_thread_stop_event, skip_cmis_mgr=False):
+        threading.Thread.__init__(self)
+        self.name = "CmisManagerTask"
+        self.exc = None
+        self.task_stopping_event = threading.Event()
```

Comment:
Can you please explain how creating a event inside this thread will be used?

Reply:
This event will be used to stop the handling with task_worker routine (in case of exception handling or graceful shutdown of the thread). The task_worker routine will be running in a while loop waiting for task_stopping_event to be set.
Hence, once we want the thread to exit, we must stop the execution of task_worker routine so that it comes out of the while loop.